enums.rb

```diff
@@ -0,0 +1,221 @@
+# rubocop:disable Style/For
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+module Enumerable
+  def my_each
+    if block_given?
+      for element in self
+        yield(element)
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      for i in (0...size)
+        yield(i, self[i])
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+
+      arr = []
+      my_each do |element|
+        arr << element if yield(element)
+      end
+      arr
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(arg = nil)
+    response = true
+    if block_given?
+      my_each do |element|
+        response = false unless yield(element)
+      end
+
+    elsif !arg.nil?
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1, false, 'hi', []].my_all? != [1, false, 'hi', []].all?
```

Reply:
Fixed this :) 
![image](https://user-images.githubusercontent.com/53732843/81595665-17a4e180-9380-11ea-904b-56933a243049.png)